flocker/control/_persistence.py

```diff
@@ -279,10 +279,13 @@ def update_leases(transform, persistence_service):
         persistence service has saved.
     """
     config = persistence_service.get()
-    new_config = config.set("leases", transform(config.leases))
-    d = persistence_service.save(new_config)
-    d.addCallback(lambda _: new_config.leases)
-    return d
+    new_leases = transform(config.leases)
+    if config.leases != new_leases:
```

Comment:
The CPU cost of doing a comparison once a second is very low. Using `is not` introduces brittleness requiring a very specific implementation, though to be fair the test should catch something that breaks this logic.